### PR TITLE
Add AWS s3 sink connector docs.

### PR DESCRIPTION
**Pull request by shibd**
### Motivation
The documentation for the current three connector cloud storage(aws-s3, Google Cloud Storage, Azure Blob Storage) is mixed together, which is not user-friendly and should be introduced separately for each cloud storage connector.

### Modifications
- Add AWS S3 sink connector docs.

### Verifying this change
- None

### Documentation

Check the box below.

Need to update docs?

- [ ] `doc-required`

  (If you need help on updating docs, create a doc issue)

- [ ] `no-need-doc`

  (Please explain why)

- [x] `doc`

  (If this PR contains doc changes)

### CODE REVIEW

File: docs/aws-s3-sink.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+| `parquetCodec`                  | String  | False    | gzip         | Compression codec used when formatType=`parquet`. Available compression types are: null (no compression), snappy, gzip, lzo, brotli, lz4, zstd.                                                                                                                                                                                                                                       |
+| `jsonAllowNaN`                  | Boolean | False    | false        | Recognize 'NaN', 'INF', '-INF' as legal floating number values when formatType=`json`. Since JSON specification does not allow such values this is a non-standard feature and disabled by default.                                                                                                                                                                                    |
+
+## Advanced features
+
+### Data format types
+
+Cloud Storage Sink Connector provides multiple output format options, including JSON, Avro, Bytes, or Parquet. The default format is JSON.
```
Comment (RobertIndie) [inline docs/aws-s3-sink.md:140]:
```suggestion
AWS S3 Sink Connector provides multiple output format options, including JSON, Avro, Bytes, or Parquet. The default format is JSON.
```

We didn't mention `Cloud storage sink connector` before. The user may not know the relationship between the cloud storage sink connector and S3 sink connector.

File: docs/aws-s3-sink.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+| `partitionerType`               | String  | True     | null         | The partitioning type. It can be configured by topic partitions or by time. By default, the partition type is configured by topic partitions.                                                                                                                                                                                                                                         |
+| `region`                        | String  | False    | null         | The AWS S3 region. Either the endpoint or region must be set.                                                                                                                                                                                                                                                                                                                         |
```
Comment (RobertIndie) [inline docs/aws-s3-sink.md:113]: Do we need to mention or link to how to get the region for the AWS?

File: docs/aws-s3-sink.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+>  PulsarIOCSCProtobufMessageMetadata __message_metadata__ = 3;
+> }
+> ```
+>
+ 
+ 
+### Dead-letter topics
+
+To use a dead-letter topic, you need to set `skipFailedMessages` to `false`, and set `--max-redeliver-count` and `--dead-letter-topic` when submit the connector with the `pulsar-admin` CLI tool. For more info about dead-letter topics, see the [Pulsar documentation](https://pulsar.apache.org/docs/en/concepts-messaging/#dead-letter-topic).
+If a message fails to be sent to the Cloud Storage and there is a dead-letter topic, the connector will send the message to the dead-letter topic.
```
Comment (RobertIndie) [inline docs/aws-s3-sink.md:199]:
```suggestion
If a message fails to be sent to the AWS S3 and there is a dead-letter topic, the connector will send the message to the dead-letter topic.
```

File: docs/aws-s3-sink.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+### Sink flushing only after batchTimeMs elapses
+
+There is a scenario where the sink is only flushing whenever the `batchTimeMs` has elapsed, even though there are many messages waiting to be processed.
+The reason for this is that the sink will only acknowledge messages after they are flushed to cloud storage but the broker stops sending messages when it reaches a certain limit of unacknowledged messages.
```
Comment (RobertIndie) [inline docs/aws-s3-sink.md:204]:
```suggestion
The reason for this is that the sink will only acknowledge messages after they are flushed to AWS S3 but the broker stops sending messages when it reaches a certain limit of unacknowledged messages.
```

File: docs/aws-s3-sink.md (hunk 2 above)
Comment (shibd) [reply docs/aws-s3-sink.md:113]: I don't think it's necessary. This should be common sense.

**Review by RobertIndie — APPROVED** [review]

**Pull request merged**